### ISSUE TITLE
Unable to start on GT-AXE16000

### ISSUE THREAD

**Issue by JonahMMay** [lynxthecat/cake-autorate#321]
When I attempt to run the script after following the AsusWRT-Merlin guide, I get the following error: `./cake-autorate.sh: line 34: syntax error: unexpected redirection`

**Comment by rany2** [comment]
Did you run with `sh` instead of `bash`?

**Comment by JonahMMay** [comment]
Yes, I ended up having to use the command `/opt/bin/bash cake-autorate.sh` despite what the documentation said. I also noticed that the install documentation says that _primary.config.sh_ should be located in _/jffs/scripts/cake-autorate_ but it is actually in _/jffs/configs/cake-autorate_.

All I believe I have left is to enable Cake which involves rebooting the router. Just waiting for someone in the house to finish a call in about a half hour to do so.

**Comment by rany2** [comment]
The script requires `bash` so that's expected. The shebang is already set properly so there's no issue there. I also didn't notice anything wrong with the documentation, can you point me to where it says that `sh` should be used?

I'll need to update the path to config for the documentation though.

**Comment by JonahMMay** [comment]
Under "Manual Testing" it says to do 
```
cd /root/cake-autorate     # to the cake-autorate directory
./cake-autorate.sh
```
but instead, I ended up having to do
```
cd /jffs/scripts/cake-autorate     # to the cake-autorate directory
/opt/bin/bash cake-autorate.sh
```

It looks like Manual Testing may just only be written for OpenWRT and not Merlin.

**Comment by rany2** [comment]
It's odd that `/usr/bin/env bash ./cake-autorate.sh` doesn't work...

**Comment by JonahMMay** [comment]
I saw in some forums where someone had a similar error on Merlin but they said running `opkg install bash` resolved the problem. Running that for me just gives `Package bash (5.2.21-1) installed in root is up to date.`

Maybe the reboot I'll be able to perform soon will fix the issue. We'll see. Launcher.sh is giving a similar syntax error on line 3 due to the opening parenthesis at the moment.

**Comment by rany2** [comment]
Mhmm, I wonder if Merlin ships with an outdated bash version and it's somehow preferring that in its PATH?

**Comment by JonahMMay** [comment]
`which bash` seems to be pointing to `/bin/bash` which is symlinked to busybox. `/usr/bin/env` appears to have a similar busybox symlink. Seems like the only way I can get the manual test to run is by manually specifying `/opt/bin/bash` in the command or by editing the shebang to use it.

**Comment by JonahMMay** [comment]
Looks like my entware installation recommended editing the PATH value which seems to be resolving the issue. Not sure why it did not do it during the install
```
Info: Congratulations!
Info: If there are no errors above then Entware was successfully initialized.
Info: Add /opt/bin & /opt/sbin to $PATH variable
Info: Add "/opt/etc/init.d/rc.unslung start" to startup script for Entware services to start
Info: Use ssh server from Entware for better compatibility.
Info: Found a Bug? Please report at https://github.com/Entware/Entware/issues
```

Just working on making it persistent then I'll be attempting that reboot.

**Comment by rany2** [comment]
> `which bash` seems to be pointing to `/bin/bash` which is symlinked to busybox

I sure hope that's not something the official busybox distribution is doing....because that's such a bad idea.

**Comment by lynxthecat** [comment]
Perhaps we can close this now @JonahMMay?
